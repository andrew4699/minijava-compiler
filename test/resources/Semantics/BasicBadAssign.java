class Factorial{
	public static void main(String[] a){
		System.out.println(new A().ComputeFac(10));
	}
}

class A {
	int a;
	boolean b;

	public int ComputeFac(int num){
		int num_aux ;
		double dub_aux ;
		num_aux = 15;

		dub_aux = 15.;
		dub_aux = 15;

    a = 5;
    b = 10;
    num = false;

		return num_aux;
	}
}
