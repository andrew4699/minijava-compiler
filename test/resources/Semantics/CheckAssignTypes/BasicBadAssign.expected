[CheckAssignTypesVisitor - 17]: Trying to assign int to double
[CheckAssignTypesVisitor - 20]: Trying to assign int to boolean
[CheckAssignTypesVisitor - 21]: Trying to assign boolean to int
